src/Clapi/TimeDiffProtocol.hs

```diff
@@ -16,6 +16,12 @@ getMonotonicTimeFloat = (\ts -> fromRational $ (toRational $ toNanoSecs ts) / 10
 
 newtype TimeDelta = TimeDelta Float
 
+tdZero :: TimeDelta
+tdZero = TimeDelta 0.0
+
+getDelta :: Time -> IO TimeDelta
+getDelta theirTime = (\now -> TimeDelta $ now - (timeToFloat theirTime)) <$> getMonotonicTimeFloat
```

Comment:
Potentially `getMonotonicTimeFloat >>= return . TimeDelta . subtract (timeToFloat theirTime)` might be clearer? (It scans better for me at any rate.)

Reply:
I think you're a flip short, but otherwise grand.

Reply:
`subtract` works backwards to how you might think: `(subtract 19) 20 == 1`. It's needed because unary negation is weird and means that you can't slice subtraction like you can for `*`, `+` and `/`. I think it's defined that way precisely so you don't need the `flip`, unless I'm not catching your drift.

Reply:
You are correct, I had assumed it was like `fmap` and `<$>` and was the non-infix version of `-`, but it isn't.